### PR TITLE
Bug 1737053: Fixing error 409 container name exists

### PR DESCRIPTION
**Pull request by coreydaley**
CreateStorage was always using a static container name of "image-registry", so if you removed the container name, it would try to recreate the same container name again.  Added some randomization to the name

### CODE REVIEW

File: pkg/storage/azure/azure.go [hunk 1]
```diff
@@ -430,53 +418,42 @@ func (d *driver) CreateStorage(cr *imageregistryv1.Config) error {
...
 		} else {
 			// TODO: do we need to create a storage account if we are provided with its name?
-			err = d.createStorageAccount(storageAccountsClient, cfg.ResourceGroup, d.Config.AccountName, cfg.Region)
+			result, err := d.accountExists(storageAccountsClient, d.Config.AccountName)
 			if err != nil {
-				if _, ok := err.(*errNameNotAvailable); !ok {
+				return err
+			}
+			if *result.NameAvailable {
+				if err = d.createStorageAccount(storageAccountsClient, cfg.ResourceGroup, d.Config.AccountName, cfg.Region); err != nil {
```
Comment (adambkaplan) [inline pkg/storage/azure/azure.go]: We also need to mark that storage is managed in this situation. My understanding is that if we are creating the storage account, we are creating the underlying storage, too.

File: pkg/storage/azure/azure.go [hunk 2]
```diff
@@ -486,23 +463,37 @@ func (d *driver) CreateStorage(cr *imageregistryv1.Config) error {
 				return err
 			}
 
-			containerName := "image-registry"
+			for i := 0; i < 5000; i++ {
+				// If the bucket name is blank, let's generate one
+				if len(d.Config.Container) == 0 {
+					d.Config.Container = fmt.Sprintf("%s-%s-%s", infra.Status.InfrastructureName, imageregistryv1.ImageRegistryName, strings.Replace(string(uuid.NewUUID()), "-", "", -1))[0:62]
```
Comment (adambkaplan) [inline pkg/storage/azure/azure.go]: nit - comment that the container name limit is 63 characters.

Comment (coreydaley) [reply pkg/storage/azure/azure.go]: updated in current push

File: pkg/storage/azure/azure.go (hunk 1 above)
Comment (coreydaley) [reply pkg/storage/azure/azure.go]: updated in current push

**Comment by coreydaley** [comment]
/retest
unrelated failures

**Pull request merged**